### ISSUE TITLE
mdMenu opens only on ng-click event due to triggerElement

### ISSUE THREAD

**Issue by GlennButera** [angular/material#4015]
**mdMenu** opens only on an *ng-click* event.  The current code specifically sets the triggerElement based on *ng-click*.

    this.init = function(setMenuContainer) {
      menuContainer = setMenuContainer;
      triggerElement = $element[0].querySelector('[ng-click]');
    };

This means that you cannot set it to open on a *ng-mouseenter* or *ng-mouseover*.  It would be better if it was using some $event means of getting the target element for the trigger instead of specifically using *ng-click*, or at the very minimum adding *ng-mouseenter* to the querySelector arguments so that menus can be triggered without clicking.

**Comment by dmackerman** [comment]
@GlennButera fork the repository, then submit a pull request with your changes.

**Comment by ThomasBurleson** [comment]
@GlennButera - The `mdMenu` triggerElement is only used to set aria attributes. Notice:

* `md-menu` creates a special scope variable `$mdOpenMenu`
*  a [custom handler/expression](https://github.com/angular/material/blob/master/src/components/menu/demoBasicUsage/index.html#L8) `ng-click="ctrl.openMenu($mdOpenMenu, $event)"`
*  openMenu uses [$mdOpenMenu($ev)](https://github.com/angular/material/blob/master/src/components/menu/demoBasicUsage/script.js#L13) to open the menu with the target element as the button.

> This means you could easily attach a `ng-mouseover="ctrl.openMenu($mdOpenMenu, $event)'`

**Comment by GlennButera** [comment]
@ThomasBurleson - The above does not work because of the code that I referenced at the top of this issue.  I submitted a fix in a pull request.  What is happening is that the menu fails to open because the triggerElement returns as null.  The triggerElement returns as null, because in the init function Angular Material currently just does a querySelector for '[ng-click]' and nothing else.

The pull request is #4020
